src/freenet/clients/fcp/FCPPluginConnectionImpl.java

```diff
@@ -0,0 +1,1255 @@
+ *   FCPPluginConnection is not strongly referenced by the client plugin anymore. If a server
+ *   plugin then tries to obtain the client connection by its UUID again (via the aforementioned
+ *   {@link PluginRespirator#getPluginConnectionByID(UUID)}, the get will fail. So if the server
+ *   plugin stores connection UUIDs, it needs no special disconnection mechanism except for
+ *   periodically trying to send a message to each client. Once obtaining the client connection by
+ *   its UUID fails, or sending the message fails, the server can opportunistically purge the UUID
+ *   from its database.
+ *   <br/>This mechanism also works for networked FCP.<br>
+ * </p></p>
+ */
+final class FCPPluginConnectionImpl implements FCPPluginConnection {
+    
+    /** Automatically set to true by {@link Logger} if the log level is set to
+     *  {@link LogLevel#DEBUG} for this class.
+     *  Used as performance optimization to prevent construction of the log strings if it is not
+     *  necessary. */
+    private static transient volatile boolean logDEBUG = false;
+
+    /** Automatically set to true by {@link Logger} if the log level is set to
+     *  {@link LogLevel#MINOR} for this class.
+     *  Used as performance optimization to prevent construction of the log strings if it is not
+     *  necessary. */
+    private static transient volatile boolean logMINOR = false;
+    
+    static {
+        // Necessary for automatic setting of logDEBUG and logMINOR
+        Logger.registerClass(FCPPluginConnectionImpl.class);
+    }
+
+
+    /**
+     * Unique identifier among all {@link FCPPluginConnection}s and FCPPluginConnectionImpls
+     * @see #getID()
+     */
+    private final UUID id = UUID.randomUUID();
+
+    /**
+     * Executor upon which we run threads of the send functions.<br>
+     * Since the send functions can be called very often, it would be inefficient to create a new
+     * {@link Thread} for each one. An {@link Executor} prevents this by having a pool of Threads
+     * which will be recycled.
+     */
+    private final Executor executor;
+
+    /**
+     * The class name of the plugin to which this FCPPluginConnectionImpl is connected.
+     * @see #getServerPluginName()
+     */
+    private final String serverPluginName;
+
+    /**
+     * The FCP server plugin to which this connection is connected.
+     * 
+     * <p>TODO: Optimization / Memory leak fix: Monitor this with a {@link ReferenceQueue} and if it
+     * becomes nulled, remove this FCPPluginConnectionImpl from the map
+     * {@link FCPConnectionHandler#pluginConnectionsByServerName}.<br/>
+     * Currently, it seems not necessary:<br/>
+     * - It can only become null if the server plugin is unloaded / reloaded. Plugin unloading /
+     *   reloading requires user interaction or auto update and shouldn't happen frequently.<br/>
+     * - It would only leak one WeakReference per plugin per client network connection. That won't
```

Comment:
(Also, this is not remotely exploitable since loading/unloading plugins is not something which is exposed to the network AFAIK.)